### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by GregoryVds**
99% sure the CI job will not work... :pray: 
I put the Uncrustify update in another commit to validate that CI is still failing properly

### CODE REVIEW

**Comment by GregoryVds** [comment]
Well, looks like I should have more faith in my .travis file :dancing_men:

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,15 @@
 language: c
 sudo: required
-dist: xenial
-install:
-    - sudo apt-get install -y cmake python3 python3-pip python3-setuptools pkg-config libcapstone-dev cppcheck curl wget check clang-tools-6.0
-    - sudo pip3 install scapy
-    - wget https://launchpad.net/~matttbe/+archive/ubuntu/ppa/+files/uncrustify_0.62-0ubuntu1~16.04~ricotz1_amd64.deb && sudo dpkg -i uncrustify_0.62-0ubuntu1~16.04~ricotz1_amd64.deb
-before_script:
-    - sudo sysctl -w net.ipv4.tcp_fastopen=5
+dist: bionic
```
Comment (matttbe) [inline .travis.yml:3]: good that it is finally available :)

**Comment by GregoryVds** [comment]
Damn, i hoped it would run a build on the first commit only. 
Triggered another build to validate failing case before merging this one: https://github.com/Tessares/libconvert/pull/16

File: README.md [hunk 2]
```diff
@@ -14,42 +14,38 @@ Future work:
...
 $ git submodule init && git submodule update
-$ mkdir build && cd build && cmake .. && make
 ```
 
-Usage (assuming a Transport Converter listening at 192.0.2.1:1234):
+The easiest way to build and run the tests is with the provided Dockerfile (which contains all deps):
 ```
-$ CONVERT_ADDR=192.0.2.1 CONVERT_PORT=1234 LD_PRELOAD=./libconvert_client.so curl https://www.tessares.net
+$ docker build -t tessares.net/libconvert .
+$ docker run --cap-add=NET_ADMIN --sysctl net.ipv4.tcp_fastopen=5 -v $PWD:/lc -t tessares.net/libconvert /bin/bash -c "mkdir /lc/build && cd /lc/build && cmake .. && make && make test"
```
Comment (matttbe) [inline README.md]: we could also add `--user "${UID}:${UID}"`

Comment (matttbe) [inline README.md]: maybe `mkdir -p`

**Review by matttbe — APPROVED** [review]
looks good!
Just two ideas, up to you ;-)

File: README.md (hunk 2 above)
Comment (GregoryVds) [reply README.md]: Good idea, will change that tomorrow

Comment (GregoryVds) [reply README.md]: Yep good idea, will change that

Comment (GregoryVds) [reply README.md]: Well, tried that, but iptables & other utils are giving me troubles inside the container as they expect to be uid 0. Don't have much time to further work on this, will merge like this then

**Pull request merged**